releng/setup-env.sh

```diff
@@ -268,6 +268,15 @@ case $host_platform in
     base_compiler_flags="-ffunction-sections -fdata-sections"
     base_linker_flags="-Wl,--gc-sections -Wl,-z,noexecstack $libgcc_flags"
 
+    case "$frida_libc" in
+      musl)
+        base_compiler_flags="$base_compiler_flags -DHAVE_MUSLLIBC"
+        ;;
+      uclibc)
+        base_compiler_flags="$base_compiler_flags -DHAVE_UCLIBC"
```

Comment:
We should not set any `HAVE_` defines in the environment. Each module should provide those as needed using their build system, such as `configure.ac` or `meson.build`. (Or sniff preprocessor defines set by the particular libc's headers.)

Reply:
I have changed the way in which I detect the CRT by using the pre-processor definitions they provide themselves.

glibc defines __GLIBC__
uclibc defines __GLIBC__ and __UCLIBC__
musl defines neither.

I have decided to use the absence of definition of __GLIBC__ or __UCLIBC__ as indication of musl. Whilst this does give the rise of potential false positives, these probably represent the 3 most common CRTs on Linux and I am not aware of any others having been supported by FRIDA to date, hence it should not introduce any regressions.

Please can you check for me that this doesn't cause any issues with android (which I believe uses bionic)?